hnn_core/cell_response.py

```diff
@@ -381,7 +385,33 @@ def plot_spikes_hist(self, trial_idx=None, ax=None, spike_types=None,
         return plot_spikes_hist(self, trial_idx=trial_idx, ax=ax,
                                 spike_types=spike_types, show=show)
 
-    def write(self, fname):
+    def _write_hdf5(self, fname):
+        """Write spiking activity to a hdf5 file.
+
+        Parameters
+        ----------
+        fname : str
+            String format (e.g., 'spk.hdf5') of the
+            path to the output spike file.
+
+        Outputs
+        -------
+        A hdf5 file containing the cell response object
+        """
+        print(f'Writing file {fname}')
+        data = dict()
+        data['spike_times'] = self.spike_times
+        data['spike_gids'] = self.spike_gids
+        data['spike_types'] = self.spike_types
+        data['vsec'] = self.vsec
+        data['isec'] = self.isec
+        data['times'] = self.times
+        # h5py cannot handle np.array(None)
+        if data['times'].shape == np.array(None).shape:
```

Comment:
when does this happen? Don't think this should ever happen ...

Reply:
do we prevent user from saving `cell_response` if simulation has not been run?

Reply:
> when does this happen? Don't think this should ever happen ...

It came up when testing saving of empty spike file.